packages/salesforcedx-vscode-core/src/commands/forceOrgCreate.ts

```diff
@@ -53,8 +62,60 @@ export class ForceOrgCreateExecutor extends SfdxCommandletExecutor<
       .withFlag('--durationdays', data.expirationDays)
       .withArg('--setdefaultusername')
       .withLogName('force_org_create_default_scratch_org')
+      .withJson()
       .build();
   }
+
+  public execute(response: ContinueResponse<AliasAndFileSelection>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+
+    channelService.streamCommandStartStop(execution);
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(realData => {
+      stdOut += realData.toString();
+    });
+
+    execution.processExitSubject.subscribe(async exitCode => {
+      this.logMetric(execution.command.logName, startTime);
+      try {
+        const createParser = new OrgCreateResultParser(stdOut);
+
+        if (createParser.createIsSuccessful()) {
+          // NOTE: there is a beta in which this command also allows users to create sandboxes
+          // once it's GA this will have to be updated
+          setWorkspaceOrgTypeWithOrgType(OrgType.SourceTracked);
+        } else {
+          const errorResponse = createParser.getResult() as OrgCreateErrorResult;
```

Comment:
Handling the scratch org creation will look something like this:

<img width="1070" alt="scratch_org_create_error" src="https://user-images.githubusercontent.com/1041842/60826936-db9f5f80-a163-11e9-9997-31cd500270a3.png">